src/Service.Tests/SqlTests/SqlTestHelper.cs

```diff
@@ -149,6 +151,65 @@ public static void TestForErrorInGraphQLResponse(string response, string message
             }
         }
 
+        /// <summary>
+        /// Validates the result from the sql db matches the graphql result returned by the engine.
+        /// </summary>
+        /// <param name="groupByArray">groupByArray</param>
+        /// <param name="expectedArray">expectedArray</param>
+        public static void AssertNumericAggregations(JsonElement groupByArray, JsonElement expectedArray, bool isfieldsPresentInResponse = true, bool isAggregatesPresentInResponse = true)
+        {
+            // Assert: Ensure expected and actual are arrays
+            Assert.AreEqual(JsonValueKind.Array, expectedArray.ValueKind);
+            Assert.AreEqual(JsonValueKind.Array, groupByArray.ValueKind);
+
+            // Convert expected values into a list of dictionaries for easy lookup
+            List<Dictionary<string, JsonElement>> expectedList = expectedArray.EnumerateArray()
+                .Select(obj => obj.EnumerateObject().ToDictionary(prop => prop.Name, prop => prop.Value))
+                .ToList();
+
+            int index = 0;
+            // Act: Iterate over each `groupBy` object in actual
+            foreach (JsonElement groupByObject in groupByArray.EnumerateArray())
+            {
+                // Create a combined dictionary and populate it with fields first
+                Dictionary<string, JsonElement> combinedDictionary = new();
+
+                if (isfieldsPresentInResponse)
+                {
+                    Assert.IsTrue(groupByObject.TryGetProperty("fields", out JsonElement fields), "Fields object not found.");
+                    Assert.AreEqual(JsonValueKind.Object, fields.ValueKind);
+
+                    // Add fields to the combined dictionary
+                    foreach (JsonProperty field in fields.EnumerateObject())
+                    {
+                        combinedDictionary[field.Name] = field.Value;
+                    }
+                }
+
+                if (isAggregatesPresentInResponse)
+                {
+                    Assert.IsTrue(groupByObject.TryGetProperty(QueryBuilder.GROUP_BY_AGGREGATE_FIELD_NAME, out JsonElement aggregations), "Aggregations object not found.");
+                    Assert.AreEqual(JsonValueKind.Object, aggregations.ValueKind);
+
+                    // Add aggregations to the combined dictionary
+                    foreach (JsonProperty aggregation in aggregations.EnumerateObject())
+                    {
+                        combinedDictionary[aggregation.Name] = aggregation.Value;
+                    }
+                }
+
+                // Convert actual aggregations and expectedList[index] to strings
+                string resultString = JsonSerializer.Serialize(combinedDictionary);
```

Comment:
if both isFieldsPresentInResponse and isAggregatesPresentInResponse are false, will the combinedDictionary be empty? Is it possible?
If its empty, what is the expectedList going to be?

Is there a scenario of empty fields which we left out to test?